### PR TITLE
[IMP] lunch: migrate to Owl

### PR DESCRIPTION
**Pull request by kbapt**
Migrate lunch to Owl.

task-2916338

### CODE REVIEW

File: addons/lunch/static/src/components/lunch_dashboard.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+}
+LunchOrderLine.template = 'lunch.LunchOrderLine';
+LunchOrderLine.props = ["line", "currency", "onUpdateQuantity", "openOrderLine"];
+LunchOrderLine.components = {
+    LunchCurrency,
+};
+
+export class LunchAlert extends Component {
+    get message() {
+        return markup(this.props.message);
```
Comment (kba-odoo) [inline addons/lunch/static/src/components/lunch_dashboard.js:57]: Hi @odoo/rd-security :wave: 

Could you have a look at this PR please? :-)

`markup()` is used here to render the content of the alert's message (which is an html field), before it was done with a t-out[1]. The alerts are coming through this controller here[2]


[1] : https://github.com/odoo/odoo/blob/3bb0aaaa0833e3d72fbb07e47e42441e293babb7/addons/lunch/static/src/xml/lunch_templates.xml#L6
[2] : https://github.com/odoo/odoo/blob/8295bd1ba0dde8688defbb2fdf6e287ef2da7273/addons/lunch/controllers/main.py#L118

Comment (mart-e) [reply addons/lunch/static/src/components/lunch_dashboard.js:57]: Why is the `message` an HTML field btw? Any lunch user can create and modify lunch alert messages, hence inject html (thankfully it's sanitized but still, not sure if necessary)

**Comment by tivisse** [comment]
@robodoo override=ci/security

Rationale: The alert creation rights are now restricted to lunch managers, so the usage of Markup is sufficiently safe as it requires manager accesses to inject HTML.

**Comment by odony** [comment]
> Rationale: The alert creation rights are now restricted to lunch managers, so the usage of Markup is sufficiently safe as it requires manager accesses to inject HTML.

Thanks for changing that :+1:

Note that the rationale about being restricted to "Lunch Administrators" only holds because the `message` field is still sanitized, as @mart-e noted. Being in charge of configuring the lunch products isn't a very trustworthy position in a company, and not equivalent to being in charge of the full website or the full accounting, for example.

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo rebase-ff

**Pull request closed without merging**